src/metabase/util/malli.cljc

```diff
@@ -124,6 +135,57 @@
      (-defn (macros/case :clj :clj :cljs :cljs)
             mx/SchematizedParams args)))
 
+#?(:clj
+   (core/defn- -fn [schema args]
+     (let [{:keys [arities meta name parglists raw-arglists schema single]} (-parse-defn-like schema args)
+           name (or name (gensym "mu-fn"))
+           id   (str (gensym "id"))]
+       `(with-meta
+          (mc/-instrument
+            {:schema ~schema
+             :report explain-fn-fail!}
+            (core/fn
+              ~name
+              ~@(-emit-arglists parglists)
+              ~@(when-not single (some->> arities val :meta vector))))
+          ~(assoc meta
+                  :raw-arglists (list 'quote raw-arglists)
+                  :schema schema
+                  :validate! id)))))
```

Comment:
I don't think this validate! metadata (or `id`) is needed since you are directly calling -instrument. 

They were being used to call instrument on the proper var (which did the wrapping). But again that is no longer needed since you are returning the fn from `-instrument` directly

Reply:
Right, they were left over from older code. Removed.